libs/types-common/src/Data/Json/Util.hs

```diff
@@ -50,3 +53,27 @@ class ToJSONObject a where
 
 instance ToJSONObject Object where
     toJSONObject = id
+
+-----------------------------------------------------------------------------
+-- toFieldName
+
+-- | Convenient helper to convert field names to use as JSON fields.
+-- it removes the prefix (assumed to be anything before an uppercase
+-- character) and lowers the first character
+--
+-- Example:
+-- newtype TeamName = TeamName { tnName :: Text }
+-- deriveJSON toFieldName ''TeamName
+--
+-- would generate {To/From}JSON instances where
+-- the field name is "name"
+
+toFieldName :: Options
+toFieldName = defaultOptions{ fieldLabelModifier = lowerFirst . dropPrefix }
+  where
+    lowerFirst :: String -> String
+    lowerFirst (x:xs) = toLower x : xs
```

Comment:
why not `map toLower` ? or even better, converting CamelCase to under_score?

Reply:
sorry, I misread this as a recursive call lowering the whole string. I see now it's not, please ignore.

Reply:
However, you brought up a good point! This did not do what was expected for API types, fixed with: https://github.com/wireapp/wire-server/commit/733a1fdf565f8b1d0fb72d305ca31218aa673e05